class PhonebookError(Exception):
    """Базовое исключение для приложения телефонного справочника.

    Используйте как общий предок для перехвата доменных ошибок.
    """


class ValidationError(PhonebookError):
    """Нарушение правил валидации входных данных (например, имя/телефон)."""


class ContactNotFound(PhonebookError):
    """Контакт с указанным идентификатором не найден в справочнике."""


class StorageError(PhonebookError):
    """Сбой на уровне хранилища при чтении или записи данных."""
